src/GitHub.Api/Application/ApiClient.cs

```diff
@@ -262,27 +269,27 @@ public ApiClient(UriString hostUrl, IKeychain keychain, IGitHubClient githubClie
             return userCache;
         }
 
-        private async Task<bool> EnsureKeychainLoaded()
+        private async Task<bool> LoadKeychainInternal()
         {
-            logger.Trace("EnsureKeychainLoaded");
+            logger.Trace("LoadKeychainInternal");
 
             if (keychain.HasKeys)
             {
                 if (!keychain.NeedsLoad)
                 {
-                    logger.Trace("EnsureKeychainLoaded: Has keys does not need load");
+                    logger.Trace("LoadKeychainInternal: Has keys does not need load");
                     return true;
                 }
 
-                logger.Trace("EnsureKeychainLoaded: Loading");
+                logger.Trace("LoadKeychainInternal: Loading");
 
                 var uriString = keychain.Connections.First().Host;
```

Comment:
We should leave a TODO here, this is assuming there's only ever going to be one login, which isn't going to be true in the long run. We should be loading the credentials that correspond to the host of the repo we're on, or alternatively all credentials that are available for all hosts we know about.